### PR TITLE
Updating code snippets in Samples and Readme

### PR DESCRIPTION
**Pull request by JonathanCrd**
Ado item 5981.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorld.md [hunk 1]
```diff
@@ -2,26 +2,32 @@
 
 >Note: Before getting started with the samples, make sure to go trough the [prerequisites](https://github.com/Azure/azure-sdk-for-net/tree/feature/mgmt-track2/sdk/resourcemanager/Azure.ResourceManager.Core#prerequisites).
 
+Namespaces for this example:
+```C# Snippet:Hello_World_Namespaces
+using System;
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorld.md:7]: don't we need Azure.ResourceManager.Core?

File: sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorldAsync.md [hunk 2]
```diff
@@ -2,28 +2,35 @@
 
 >Note: Before getting started with the samples, go through the [prerequisites](https://github.com/Azure/azure-sdk-for-net/tree/feature/mgmt-track2/sdk/resourcemanager/Azure.ResourceManager.Core#prerequisites).
 
+Namespaces for this example:
+```C# Snippet:Hello_World_Async_Namespaces
+using System;
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorldAsync.md:7]: don't we need Azure.ResourceManager.Core?

File: sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample2_ManagingResourceGroups.md [hunk 3]
```diff
@@ -1,16 +1,22 @@
 Example: Managing Resource Groups
 --------------------------------------
+For this example, you need the following namespaces:
+```C# Snippet:Managing_Resource_Groups_Namespaces
+using System;
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample2_ManagingResourceGroups.md:5]: don't we need Azure.ResourceManager.Core?

File: sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample3_CreatingAVirtualNetwork.md [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
 --------------------------------------
 
-In this example, we'll create a virtual network. Since the SDK follows the resource hierarchy in Azure, we'll need to do this inside of a resource group. Start by creating a new resource group, like we did above:
+In this example, we'll create a virtual network. Since the SDK follows the resource hierarchy in Azure, we'll need to do this inside of a resource group.
 
-```csharp
+## Import the namespaces
+These are the naspaces needed for this project:
+```C# Snippet:Creating_A_Virtual_Network_Namespaces
+using System.Threading.Tasks;
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample3_CreatingAVirtualNetwork.md:9]: don't we need Azure.ResourceManager.Core and Azure.ResourceManager.Network?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorld.md (hunk 1 above)
Comment (JonathanCrd) [reply sdk/resourcemanager/Azure.ResourceManager.Core/samples/Sample1_HelloWorld.md:7]: Yes, good catch.

File: sdk/resourcemanager/Azure.ResourceManager.Core/README.md [hunk 5]
```diff
@@ -33,14 +33,15 @@ The default option to create an authenticated client is to use `DefaultAzureCred
...
 using System;
-    
-// code omitted for brevity
-    
-var armClient = new ArmClient(new DefaultAzureCredential());
+using System.Threading.Tasks;
+
+            // code omitted for brevity
+
+            var armClient = new ArmClient(new DefaultAzureCredential());
```
Comment (scottaddie) [inline sdk/resourcemanager/Azure.ResourceManager.Core/README.md:44]: Dedent this code so that it lines up with the `using` statements.

**Review by m-nash — APPROVED** [review]

**Pull request merged**